tests/ui/test_settings_enable_2fa.py

```diff
@@ -0,0 +1,43 @@
+import re
+
+from playwright.sync_api import Page, expect
+from tenacity import RetryCallState, retry, stop_after_attempt, wait_random_exponential
+
+BASE_URL = "http://127.0.0.1:8080"
+
+
+def log_retry_error(retry_state: RetryCallState) -> None:
+    if retry_state is None or retry_state.outcome is None:
+        return
+    print(f"Retrying due to error: {retry_state.outcome.exception()}")
+
+
+@retry(
+    wait=wait_random_exponential(min=1, max=60),
+    stop=stop_after_attempt(3),
+    retry_error_callback=log_retry_error,
+)
+def test_enable_2fa_has_back_button(page: Page, user_password: str) -> None:
+    page.goto(f"{BASE_URL}/login", wait_until="domcontentloaded")
+    page.get_by_label("Username").fill("test")
+    page.get_by_label("Password").fill(user_password)
+    page.get_by_role("button", name="Login").click()
```

Comment:
i tried moving this out into a `pytest` fixture for the test methods to use before testing the actual button-related changes but couldn't get the tests to synchronously wait for a successful login 🤔 so there's duplication here and test below which isn't great but in essence of time, could still be a good first pass at integrating some browser testing.